metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -2031,4 +2043,21 @@ private DataPlatformInfo getDataPlatformInfo(Urn urn) {
     }
     return null;
   }
+
+  private void checkConditionalUpdate(EntityAspect oldAspect, ExtraIngestParams extraIngestParams) {
+    Long updateIfCreatedOn = extraIngestParams != null && oldAspect != null
+            ? extraIngestParams.getCreatedOn(oldAspect.getUrn(), oldAspect.getAspect()) : null;
+    if (_aspectDao.supportTransactions() && updateIfCreatedOn != null) {
+      long updateIfCreatedOnValue = updateIfCreatedOn.longValue();
+      if (oldAspect != null && updateIfCreatedOnValue == 0) {
+        throw new PreconditionFailedException("Conditional Update value must be greater then 0 for update operation");
+      }
+      if (oldAspect == null && updateIfCreatedOnValue != 0) {
+        throw new PreconditionFailedException("Conditional Update value must be 0 for insert operation");
+      }
+      if (updateIfCreatedOnValue != 0 && oldAspect != null && oldAspect.getCreatedOn().getTime() != updateIfCreatedOnValue) {
+        throw new PreconditionFailedException("Condition for Conditional Update is not met");
```

Comment:
Example here: What URN+aspect are we updating? Why couldn't we update it? What is the expected `createdOn` value? What was the actual `createdOn` value? (Who was it changed by?)